docs/source/configuration/slots.md

```diff
@@ -0,0 +1,260 @@
+    });
+```
+
+A slot component must have the following parameters.
+
+`slot`
+:   The name of the slot, where the slot components are stored.
+
+`name`
+:   The name of the slot component that we are registering.
+
+`component`
+:   The component that we want to render in the slot.
+
+`predicates`
+:   A list of functions that return a function with this signature.
+
+    ```ts
+    export type SlotPredicate = (args: any) => boolean;
+    ```
+
+
+## Predicate helpers
+
+There are two predicate helpers available in the Volto helpers.
+
+
+### `RouteCondition`
+
+```ts
+export function RouteCondition(path: string, exact?: boolean) {
+  return ({ pathname }: { pathname: string }) =>
+    Boolean(matchPath(pathname, { path, exact }));
+}
+```
+
+The `RouteCondition` predicate helper renders a slot if the specified route matches.
+It accepts the following parameters.
+
+`path`
+:   String.
+    Required.
+    The route.
+
+`exact`
+:   Boolean.
+    Optional.
+    If `true`, then the match will be exact, else matches "begins with", for the given string from `path`.
+
+
+### `ContentTypeCondition`
+
+```ts
+export function ContentTypeCondition(contentType: string[]) {
+  return ({ content }: { content: Content }) =>
+    contentType.includes(content['@type']);
+}
+```
+
+The `ContentTypeCondition` helper predicate allows you to render a slot when the given content type matches the current content type.
+It accepts a list of possible content types.
+
+
+### Custom predicates
+
+You can create your own predicate helpers to determine whether your slot component should render.
+The `SlotRenderer` will pass down the current `content` and the `pathname` into your custom predicate helper.
+You can also tailor your own `SlotRenderer`s, or shadow the original `SlotRenderer`, to satisfy your requirements.
+
+
+## Manage registered slots and slot components
+
+You can manage registered slots and slot components through the slots API.
+
+
+### `getSlotComponents`
+
+`getSlotComponents` returns the list of components registered per slot.
+This is useful to debug what is registered and in what order, informing you whether you need to change their order.
+This is the signature:
+
+```ts
+config.getSlotComponents(slot: string): string[]
+```
+
+`slot`
+:   String.
+    Required.
+    The name of the slot, where the slot components are stored.
+
+
+### `reorderSlotComponent`
+
+`reorderSlotComponent` reorders the list of slot components registered per slot.
+This is the signature:
+
+```ts
+config.reorderSlotComponent(slot: string, name: string, position: number): void
+```
+
+`slot`
+:   String.
+    Required.
+    The name of the slot where the slot components are stored.
+
+`name`
+:   String.
+    Required.
+    The name of the slot component to reposition in the list of slot components.
+
+`position`
+:   Number.
+    Required.
+    The destination position in the registered list of slot components.
+    The position is zero-indexed.
```

Comment:
It'd be nice to provide an easy way to put a name before/after another name, or at the end of the list.

Reply:
This is next.

Reply:
Done.